crates/datasources/src/excel/mod.rs

```diff
@@ -52,9 +53,15 @@ impl ExcelTable {
 
             DatasourceUrl::Url(_) => {
                 let store = store_access.create_store()?;
-                let list = store_access
-                    .list_globbed(&store, source_url.path().as_ref())
-                    .await?;
+
+                let path = source_url.path().into_owned();
+                let paths = get_resolved_patterns(path);
+
+                let mut list = Vec::new();
+                for path in paths {
+                    let sub_list = store_access.list_globbed(&store, path).await?;
+                    list.extend(sub_list);
+                }
 
```

Comment:
does it make sense to have a helper function to wrap all this up into a single operation? we seem to do it three times...

Reply:
Yeah, we should have a helper function do this.